lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -514,7 +519,12 @@ private function loadProxyClass(ClassMetadata $class): string
                 }
                 // no break
             case self::AUTOGENERATE_ALWAYS:
-                $this->generateProxyClass($class, $fileName, $proxyClassName);
+                $handle = fopen(sys_get_temp_dir() . '/generate_proxy_class.lock', 'w+');
```

Comment:
Is `w+` really needed?

Reply:
Which mode would you suggest? The file `sys_get_temp_dir() . '/generate_proxy_class.lock` has to exists in order to create lock.

Reply:
I made that suggestion because I took a look at the first example of the [`flock` docs](https://www.php.net/manual/en/function.flock.php). They use `r+`

Reply:
But you're right, I tried running the example, and it seems to suppose the file exists.

Reply:
I had it same. Anyway, I changed it to `wb` which should be also sufficient.